### PR TITLE
PIM-6338: Enrich product model (save part)

### PR DESCRIPTION
**Pull request by grena**
**Description (for Contributor and Core Developer)**

This PR is a part of PIM-6338 (enrich product model) and is about the ability to save the product model when we edit it through the PEF.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | N
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | N
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductModelController.php [hunk 1]
```diff
@@ -19,33 +26,76 @@
...
+
+    /** @var AttributeConverterInterface */
+    private $localizedConverter;
+
+    /** @var EntityWithValuesFilter */
+    private $emptyValuesFilter;
+
+    /** @var ConverterInterface */
+    private $productValueConverter;
+    /**
```
Comment (SamirBoulil) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductModelController.php]: one line

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product-model/form/save.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        BaseSave,
+        messenger,
+        ProductManager,
+        ProductModelSaver,
+        FieldManager,
+        i18n,
+        UserContext
+    ) {
+        return BaseSave.extend({
+            updateSuccessMessage: __('pim_enrich.entity.product_model.info.update_successful'),
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product-model/form/save.js]: should be conf

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product-model/form/save.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
+'use strict';
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product-model/form/save.js]: it should be the same module as product saver. We should only change the configuration

Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product-model/form/save.js]: I'm not really confident with that
a product model is not the same that a product 
really
they are close, that's true, and they have some common behaviors but they are not the same

we are in the beginning of our implementation, it's safer imo to have 2 distincts objects
but I agree that, if they are really almost the same, we should merge them

Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product-model/form/save.js]: I know there is a lot of common code, but as for now, I don't know if we'll need to do other special stuff. And we delete other things here that we keep in the product:
```
                delete productModel.meta;
                delete productModel.family;
                delete productModel.associations;
```

**Pull request merged**